### PR TITLE
add .circleci/config.yml

### PR DESCRIPTION
**Pull request by nonsense**
This PR is adding an initial `.circleci/config.yml` which should:
1. Checkout latest `master` from `github.com/filecoin-project/lotus`
2. Build `lotus`, `lotus-seed` and `lotus-miner`
3. Checkout a given commit from `dealbot`
4. Run `go test` and integration tests.

### CODE REVIEW

**Comment by willscott** [comment]
this won't run the integration test yet, right?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - run: cd /tmp/lotus && git submodule sync && git submodule update --init
+      - run: cd /tmp/lotus && make debug && sudo make install
+      - run: cd /tmp/lotus && sudo cp lotus-seed /usr/local/bin
+  test-dealbot:
+    steps:
+      - run: go build -o ./devnet-binary cmd/devnet/*.go
+      - run: go build -o dealbot-binary
+      - run: sudo cp devnet-binary /usr/local/bin/devnet
+      - run: sudo cp dealbot-binary /usr/local/bin/dealbot
+      - run: ./integration_tests/01_storage_retrieval_ok.sh
```
Comment (nonsense) [inline .circleci/config.yml:33]: We might need to give a bit more than 30sec. for the devnet to start up when running on CI. At least on the `Docker Medium` VM I am getting on my personal fork, it takes longer - we should review once this is merged and running on `2xlarge`.

**Comment by nonsense** [comment]
> this won't run the integration test yet, right?

@willscott it does now.

---

This can certainly be made a bit nicer, and there is room for optimisation, but as a first stab, it seems to start the test.

**Review by willscott — APPROVED** [review]

**Comment by nonsense** [comment]
@willscott marking this for review, as it is not clear if when tests run on `2xlarge` we'll see the race condition on the first storage deal - I might need to adjust the integration test for that.

It takes about 4min30sec. for the `dealbot` to run a storage deal, so hopefully this is not too slow for us.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - image: circleci/golang:1.16.3
+    resource_class: 2xlarge
+
+commands:
+  checkout-dealbot:
+    steps:
+      - checkout
+  install-deps:
+    steps:
+      - go/install-ssh
```
[the remark this answers is not in the document]
Comment (nonsense) [reply .circleci/config.yml:17]: I am not sure if I can (or cannot) attach with SSH when this is missing. Will give it a try without it.

**Comment by willscott** [comment]
i'm okay with that latency on CI i think

**Comment by nonsense** [comment]
> i'm okay with that latency on CI i think

Well, there is another 1-2min for building `lotus`, which we could get rid off if we cache the binaries, but I'd rather add these one step at a time.

**Pull request merged**